### PR TITLE
Resolve Milling Issues/Errors

### LINKED ISSUE #76334: invalid recipe id "meal_bone_mill...
### Describe the bug

`E`xamine the `tiny canine skull`
```
 DEBUG    : invalid recipe id "meal_bone_mill_2_1"

 FUNCTION : const T& string_id<T>::obj() const [with T = recipe]
 FILE     : src/recipe_dictionary.cpp
 LINE     : 66
 VERSION  : cdda-experimental-2024-09-10-0339 c73e4c6
```
`E`xamine the `small canine skull`
```
 DEBUG    : invalid recipe id "meal_bone_mill_1_1"

 FUNCTION : const T& string_id<T>::obj() const [with T = recipe]
 FILE     : src/recipe_dictionary.cpp
 LINE     : 66
 VERSION  : cdda-experimental-2024-09-10-0339 c73e4c6
```

### Attach save file

n/a

### Steps to reproduce

Kill and butcher a dachshund (small) and a dachshund puppy (tiny). Check the skull

### Expected behavior

n/a

### Screenshots

_No response_

### Versions and configuration

- OS: Windows
    - OS Version: 10.0.19043 (21H1)
- Game Version: cdda-experimental-2024-09-10-0339 c73e4c6 [64-bit]
- Graphics Version: Tiles
- Game Language: English [en]
- Mods loaded: [
    Dark Days Ahead [dda],
    Disable NPC Needs [no_npc_food],
    Portal Storms Ignore NPCs [personal_portal_storms],
    Slowdown Fungal Growth [no_fungal_growth]
]

### Additional context

Skull spawning causes the game to crash
- #76332

### PR DESCRIPTION
**Pull request by TheMurderUnicorn**
#### Summary
Bugfixes "Resolve Milling Issues/Errors"

#### Purpose of change
When harvesting certain skulls/debug spawning them, errors were received stating they had missing milling recipes.  I have added these milling recipes, as well as some skull requirement groups. 

#### Describe the solution
This solution was pulled from #76334 which outlined a solution for two of the three errors.  Most of what I've done here is exactly what was outlined.  I added two requirement groups so that we can call similarly sized skulls from the milling recipes.  Then I added the milling recipes.  2 small skulls = 1 bone meal, 1 medium skull = 1 bone meal, and 20 rodent skulls = 1 bone meal.

Just as a heads up, the small canine skull has the id with 'medium' in it, and the tiny canine skull has an id with 'small' in it, these are not issues with what I've done here, they are issues with their names and ids not matching up.

#### Describe alternatives you've considered
Making Karol do it.

Debated changing the name/ids of the skulls so they match but that's out of scope and I was cautioned against fucking with IDs.  Although now that I look at it, it looks like these would be name changes not ID changes.

#### Testing
Once I wrapped my head around how milling works, I basically just did the changes and made sure that the mill took the right number of skulls and that it output the right amount of bone meal.  I also tested to see if this resolved the errors, which it seems to have fixed correctly.  Additionally, I originally got the error from #76332 but did not get it again after making my changes, though I can't tell if this is just me or if I also fixed that issue.

#### Additional context
closes #76334 

Possibly is also a fix for #76332 but I'm hesitant to say that when I don't fully understand that cause.  To my understanding it's an imgui failure when trying to output an in-game error message -- that error may have been related to the lack of a milling recipe, and so it's conceivable it was fixed here.  Again, I'm no coder, barely know what I'm doing, so I'm not going to mark this as closed when I have no idea if it's actually fixed.

Was going to ping Karol but let's be honest, your ass is already looking at this.

And here's my TLDR/checklist in case someone wants a quickly readable version
image

### CODE REVIEW

**Comment by TheMurderUnicorn** [comment]
I'll be honest, I'm very confused why the test failed and why it's even testing these skulls as having calories.  The skulls have no calories, but the bone meal does.  But also the bone meal's calories fall in line with the tests calorie bounds, so I have no idea what any of this means and will probably need a little help understanding. 

This is the Cataclysm Windows build/ Build (pull_request) failure and idk what to do about it.
image
The changes I've made are basically identical copies to existing recipes that don't trigger this, and I'm not seeing anything that indicates the skulls are somehow wrongly identified as food or anything? Anyone know what I should be doing here?

**Comment by AudBobb** [comment]
Does it need the force override tag for calories, since the ingredients have 0, causing the output to have 0, when it is expecting output to have non-zero?

Just a wild guess

**Comment by TheMurderUnicorn** [comment]
Unfortunately, Nutrient override flag can only be applied to the item itself (bone meal) afaik.  This can't be done because the other bone meal recipes use different items that do have inherent calories, and this would stop bonemeal from inheriting calories from those items.  If there's a flag that's usable to prevent recipes from pushing their calories or adds them to the base calories, I couldn't find it.  So, unless we want all bonemeal clamped at 35 calories regardless of source, I don't know how to do this.

**Comment by AudBobb** [comment]
Hmm didn't realize that flag was applied directly to the item. Looking through the other milling recipes related to bones, the bone skewer to bone meal recipe also turns an item that has no calories into one that does, and follows the exact format as your entries. I loaded up a test world and spawned a windmill, and bone skewers, and the bone meal it produces does in fact have 0 calories. 

I think the solution to this issue is to modify the ingredient items to include calories, like bone does.

**Comment by TheMurderUnicorn** [comment]
Looking at previous PRs I don't see why the skewers didn't trigger the same test issue, there was some discussion in the initial PR but I'm not really understanding it.  If it's a matter of adjusting skull calories (which it shouldn't be imo) that's above what I'll be doing.  I don't see why them inheriting 0 calories actually matters at all and it's my opinion that the test should not care if they output 0 calories unless I'm missing something glaring.

**Comment by AudBobb** [comment]
It's not a major issue tbh, the test is more a sanity check for actual food items, this is just a weird edge case. I can go in and fix the skulls real quick to have calories so that it fixes your issue. Just submitted a fix for the bone skewers, so I'm in the files anyways

**Comment by Karol1223** [comment]
Give them nutrient override. Bone meal has values based off irl bone meal, it can have that.

**Comment by TheMurderUnicorn** [comment]
This does not work, I already tried it when it was suggested earlier.  It produces bone meal with 0 calories even with NUTRIENT_OVERRIDE
image
Unless I put it in the wrong place somehow:
image

**Comment by Karol1223** [comment]
You did. It's an item flag, not a recipe flag. You put it in the item definition.

**Comment by TheMurderUnicorn** [comment]
This makes me nervous, I thought there was a reason I didn't want to clamp bonemeal to 35.  It has so many sources and I'm worried it'll mess something up but honestly I turned what I thought was a simple PR into a shitshow so whatever, this should do it.

**Comment by Karol1223** [comment]
> This makes me nervous, I thought there was a reason I didn't want to clamp bonemeal to 35. It has so many sources and I'm worried it'll mess something up but honestly I turned what I thought was a simple PR into a shitshow so whatever, this should do it.

This will nerf manual bone meal crafting recipes and tbh that's fine. You won't get 100% of the calories out of a bone if you grind it into powder, because you'll lose some powder along the way and all that. This way hand-crafted bonemeal will go from being ~43kcal a piece to 35kcal a piece. That is fine.

**Comment by Karol1223** [comment]
Oh god, nevermind, I really put you on a trap with asking you to do this one for me.

``NUTRIENt_OVERRIDE`` will break cannibalism from human bones. This is a mess.

**Comment by AudBobb** [comment]
Feels like we need a flag for recipes that forces the recipe output to have the items default calories.

**Comment by Karol1223** [comment]
> Feels like we need a flag for recipes that forces the recipe output to have the items default calories.

I'd even go as far as to say that this should be automatic and not handled by a flag - if a comestible is crafted from items that cannot have calories at all (so no comestibles as ingredients), it should be crafted with its default values instead of trying to inherit something that doesn't exist.

**Comment by TheMurderUnicorn** [comment]
When you think you can be productive ;-; Closing PR

**Comment by RenechCDDA** [comment]
> > Feels like we need a flag for recipes that forces the recipe output to have the items default calories.
> 
> I'd even go as far as to say that this should be automatic and not handled by a flag - if a comestible is crafted from items that cannot have calories at all (so no comestibles as ingredients), it should be crafted with its default values instead of trying to inherit something that doesn't exist.

...Remind me for this

**Comment by Karol1223** [comment]
Wake up, Murder, fix for this just got merged

File: data/json/items/resources/bone.json [hunk 1]
```diff
@@ -85,7 +85,7 @@
     "vitamins": [ [ "calcium", "5294 mg" ], [ "meat_allergen", 1 ] ],
     "material": [ "powder" ],
     "volume": "125 ml",
-    "flags": [ "EDIBLE_FROZEN" ],
+    "flags": [ "EDIBLE_FROZEN", "NUTRIENT_OVERRIDE" ],
```
Comment (Karol1223) [inline data/json/items/resources/bone.json:88]:
```suggestion
    "flags": [ "EDIBLE_FROZEN" ],
```

It must be done

**Review by Karol1223 — CHANGES_REQUESTED** [review]

**Comment by Karol1223** [comment]
> Yeah, so, I saw you kicked the tests, unfortunately, this is still failing due to the test being unhappy that the expected calories don't fall in line properly, and as far as I know, this has not been resolved/I don't know how to fix it despite other people having contributed to 'fix' it. So, the PR is dead until something changes and I'm not the one to resolve these issues. Should I just close the PR?

Everything is evil. I have no clue what's going on anymore. According to all known laws of aviation, the tests should not be failing, but the tests fail anyway because they don't care.
